akka-actor/src/main/scala-2.13/akka/dispatch/internal/ScalaBatchable.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2009-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.dispatch.internal
+
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object ScalaBatchable {
+
+  /*
+   * In Scala 2.13.0 `OnCompleteRunnable` was deprecated, and in 2.13.4
+   * `impl.Promise.Transformation` no longer extend the deprecated `OnCompleteRunnable`
+   * but instead `scala.concurrent.Batchable` (which anyway extends `OnCompleteRunnable`).
+   * On top of that we have or own legacy version `akka.dispatch.Batchable`.
+   */
+
+  def isBatchable(runnable: Runnable): Boolean = runnable match {
+    case b: akka.dispatch.Batchable    => b.isBatchable
+    case _: scala.concurrent.Batchable => true
```

Comment:
I'm sure I fully understand what happened to the Scala internals as described in the comment above. What if this version of Akka is run in an application with Scala 2.13.3 dependency? Will it be caught by this `scala.concurrent.Batchable` case, or will this be a regression compared to previous OnCompleteRunnable?

We might not care much about that and we can say that they should update to Scala 2.13.4. We already have another reason for that: https://github.com/akka/akka/issues/30018

Reply:
`OnCompleteRunnable` extends `Batchable` in all 2.13.x as far as I can see, so it should not cause a regression in older 2.13 releases.

Reply:
ok, thanks